### PR TITLE
HTTP headers fix, two small pre-CORS steps.

### PR DESCRIPTION
**Pull request by dkorolev**
Hi @mzhurovich,

Here's what I have hanging from the weekend. Considering leaving it as is for now. What do you think? :)

CC @sompylasar 

Thanks,
Dima

### CODE REVIEW

**Comment by mzhurovich** [comment]
Hmmm
```
[ RUN      ] PosixHTTPServerTest.InvalidHEXAsChunkSizeDoesNotKillServer

libc++abi.dylib: terminating

make[1]: *** [test] Abort trap: 6
```

File: Bricks/net/http/constants.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+#include "../../../Bricks/strings/strings.h"
+
+// HTTP constants to parse the header and extract method, URL, headers and body.
+
+namespace current {
+namespace net {
+namespace constants {
+
+constexpr static const char kCRLF[] = "\r\n";
```
Comment (mzhurovich) [inline Bricks/net/http/constants.h]: I suppose `static` is redundant here.

File: Bricks/net/http/constants.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+constexpr const size_t kCRLFLength = strings::CompileTimeStringLength(kCRLF);
```
Comment (mzhurovich) [inline Bricks/net/http/constants.h]: `constexpr size_t`

Comment (mzhurovich) [inline Bricks/net/http/constants.h]: Here and below.

File: Bricks/net/http/constants.h [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+constexpr const char* kDefaultJSONContentType = "application/json; charset=utf-8";
+// TODO(dkorolev): Make use of this constant everywhere.
+constexpr const char* kDefaultHTMLContentType = "text/html; charset=utf-8";
+constexpr const char* kDefaultSVGContentType = "image/svg+xml; charset=utf-8";
+constexpr const char* kDefaultPNGContentType = "image/png";
+
+constexpr const char kHeaderKeyValueSeparator[] = ": ";
+constexpr const size_t kHeaderKeyValueSeparatorLength = strings::CompileTimeStringLength(kHeaderKeyValueSeparator);
+
+constexpr const char* const kContentLengthHeaderKey = "Content-Length";
```
Comment (mzhurovich) [inline Bricks/net/http/constants.h]: Second `const` is redundant, `constexpr const char*` is enough.

**Comment by dkorolev** [comment]
```
.././Blocks/HTTP/docu/client/../../../../Bricks/net/http/impl/../constants.h:39:39: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 constexpr char* kDefaultContentType = "text/plain";
```

But I did make it better. And did move to c++14. PTAL.

**Comment by mzhurovich** [comment]
@dkorolev The only thing that bothers me here is the switch to `c++14`. Do you still think that we ought to do it right now?

**Comment by dkorolev** [comment]
@mzhurovich PTAL

File: scripts/check-headers-within-dir.sh [hunk 4]
```diff
@@ -27,21 +27,23 @@ rm -f "$PWD"/.current_*.cc "$PWD"/.current_*.o "$PWD"/.current_.txt "$SOURCE_FIL
...
-
-  ln -sf "$PWD/$i" "$HEADER_GCC"
-  ln -sf "$PWD/$i" "$HEADER_CLANG"
-
-  g++ -I "$EXTRA_INCLUDE_DIR" $CPPFLAGS -c "$HEADER_GCC" -o "${HEADER_GCC}.o" $LDFLAGS \
-    >"$TMP_STDOUT" 2>"$TMP_STDERR" || (cat "$TMP_STDOUT" "$TMP_STDERR" && exit 1)
-  clang++ -I "$EXTRA_INCLUDE_DIR" $CPPFLAGS -c "$HEADER_CLANG" -o "${HEADER_CLANG}.o" $LDFLAGS \
-    >"$TMP_STDOUT" 2>"$TMP_STDERR" || (cat "$TMP_STDOUT" "$TMP_STDERR" && exit 1)
-  ld -r "${HEADER_GCC}.o" "${HEADER_CLANG}.o" -o "${COMBINED_OBJECT}.o"
+  if [ "$(head -n 1 "$i")" != "// Autogenerated schema inferred from input JSON data." ] ; then
```
Comment (mzhurovich) [inline scripts/check-headers-within-dir.sh]: What do you think about more generic
`if !(head -n 1 "$i" | grep -i "auto" > /dev/null) ; then`
?

Comment (dkorolev) [inline scripts/check-headers-within-dir.sh]: :+1: c0dd893

**Comment by mzhurovich** [comment]
@dkorolev Thanks! LGTM!

**Pull request merged**